cluster/gce/coreos/helper.sh

```diff
@@ -75,38 +75,38 @@ NUM_MINIONS: $(yaml-quote ${NUM_MINIONS})
-SERVER_BINARY_TAR_URL=${SERVER_BINARY_TAR_URL}
-SERVICE_CLUSTER_IP_RANGE=${SERVICE_CLUSTER_IP_RANGE}
-ENABLE_CLUSTER_MONITORING=${ENABLE_CLUSTER_MONITORING:-none}
-ENABLE_CLUSTER_LOGGING=${ENABLE_CLUSTER_LOGGING:-false}
-ENABLE_CLUSTER_UI=${ENABLE_CLUSTER_UI:-false}
-ENABLE_NODE_LOGGING=${ENABLE_NODE_LOGGING:-false}
-LOGGING_DESTINATION=${LOGGING_DESTINATION:-}
-ELASTICSEARCH_LOGGING_REPLICAS=${ELASTICSEARCH_LOGGING_REPLICAS:-}
-ENABLE_CLUSTER_DNS=${ENABLE_CLUSTER_DNS:-false}
-ENABLE_CLUSTER_REGISTRY=${ENABLE_CLUSTER_REGISTRY:-false}
-NUM_MINIONS=${NUM_MINIONS}
-DNS_REPLICAS=${DNS_REPLICAS:-}
-DNS_SERVER_IP=${DNS_SERVER_IP:-}
-DNS_DOMAIN=${DNS_DOMAIN:-}
-KUBELET_TOKEN=${KUBELET_TOKEN:-}
-KUBE_PROXY_TOKEN=${KUBE_PROXY_TOKEN:-}
-ADMISSION_CONTROL=${ADMISSION_CONTROL:-}
-MASTER_IP_RANGE=${MASTER_IP_RANGE}
-KUBERNETES_MASTER_NAME=${MASTER_NAME}
-ZONE=${ZONE}
-EXTRA_DOCKER_OPTS=${EXTRA_DOCKER_OPTS:-}
-PROJECT_ID=${PROJECT}
-KUBERNETES_CONTAINER_RUNTIME=${CONTAINER_RUNTIME}
-RKT_VERSION=${RKT_VERSION}
-KUBERNETES_CONFIGURE_CBR0=${KUBERNETES_CONFIGURE_CBR0:-true}
-CA_CERT=${CA_CERT_BASE64}
-KUBELET_CERT=${KUBELET_CERT_BASE64:-}
-KUBELET_KEY=${KUBELET_KEY_BASE64:-}
+KUBERNETES_MASTER: $(yaml-quote "false")
+ENV_TIMESTAMP: $(yaml-quote $(date -u +%Y-%m-%dT%T%z))
+INSTANCE_PREFIX: $(yaml-quote ${INSTANCE_PREFIX})
+NODE_INSTANCE_PREFIX: $(yaml-quote ${NODE_INSTANCE_PREFIX})
+SERVER_BINARY_TAR_URL: $(yaml-quote ${SERVER_BINARY_TAR_URL})
+SERVICE_CLUSTER_IP_RANGE: $(yaml-quote ${SERVICE_CLUSTER_IP_RANGE})
+ENABLE_CLUSTER_MONITORING: $(yaml-quote ${ENABLE_CLUSTER_MONITORING:-none})
+ENABLE_CLUSTER_LOGGING: $(yaml-quote ${ENABLE_CLUSTER_LOGGING:-false})
+ENABLE_CLUSTER_UI: $(yaml-quote ${ENABLE_CLUSTER_UI:-false})
+ENABLE_NODE_LOGGING: $(yaml-quote ${ENABLE_NODE_LOGGING:-false})
+LOGGING_DESTINATION: $(yaml-quote ${LOGGING_DESTINATION:-})
+ELASTICSEARCH_LOGGING_REPLICAS: $(yaml-quote ${ELASTICSEARCH_LOGGING_REPLICAS:-})
+ENABLE_CLUSTER_DNS: $(yaml-quote ${ENABLE_CLUSTER_DNS:-false})
+ENABLE_CLUSTER_REGISTRY: $(yaml-quote ${ENABLE_CLUSTER_REGISTRY:-false})
+NUM_MINIONS: $(yaml-quote ${NUM_MINIONS})
+DNS_REPLICAS: $(yaml-quote ${DNS_REPLICAS:-})
+DNS_SERVER_IP: $(yaml-quote ${DNS_SERVER_IP:-})
+DNS_DOMAIN: $(yaml-quote ${DNS_DOMAIN:-})
+KUBELET_TOKEN: $(yaml-quote ${KUBELET_TOKEN:-})
+KUBE_PROXY_TOKEN: $(yaml-quote ${KUBE_PROXY_TOKEN:-})
+ADMISSION_CONTROL: $(yaml-quote ${ADMISSION_CONTROL:-})
+MASTER_IP_RANGE: $(yaml-quote ${MASTER_IP_RANGE})
+KUBERNETES_MASTER_NAME: $(yaml-quote ${MASTER_NAME})
+ZONE: $(yaml-quote ${ZONE})
+EXTRA_DOCKER_OPTS: $(yaml-quote ${EXTRA_DOCKER_OPTS:-})
+PROJECT_ID: $(yaml-quote ${PROJECT})
+KUBERNETES_CONTAINER_RUNTIME: $(yaml-quote ${CONTAINER_RUNTIME})
+RKT_VERSION: $(yaml-quote ${RKT_VERSION})
+KUBERNETES_CONFIGURE_CBR0: $(yaml-quote ${KUBERNETES_CONFIGURE_CBR0:-true})
+CA_CERT: $(yaml-quote ${CA_CERT_BASE64})
+KUBELET_CERT: $(yaml-quote ${KUBELET_CERT_BASE64:-})
+KUBELET_KEY: $(yaml-quote ${KUBELET_KEY_BASE64:-})
```

Comment:
As you move this back to yaml format, why not just use the ContainerVM code and then append whatever dedicated for coreos? That will remove unnecessary code and reduce maintenance cost.